### PR TITLE
Extending the discouraged functions list

### PR DESCRIPTION
**Pull request by Nicasso**
Adding ksm, kint, and ddbug_backtrace to the list of discouraged functions.

Related to these 3 issues on Drupal.org

https://www.drupal.org/project/coder/issues/3129320.
https://www.drupal.org/project/coder/issues/3129343
https://www.drupal.org/project/coder/issues/3129344

### CODE REVIEW

File: coder_sniffer/Drupal/Sniffs/Functions/DiscouragedFunctionsSniff.php [hunk 1]
```diff
@@ -34,6 +34,7 @@ class DiscouragedFunctionsSniff extends ForbiddenFunctionsSniff
         'dargs'               => null,
         'dcp'                 => null,
         'dd'                  => null,
+        'ddbug_backtrace'     => null,
```
Comment (klausi) [inline coder_sniffer/Drupal/Sniffs/Functions/DiscouragedFunctionsSniff.php]: typo, should be ddebug_backtrace

**Review by klausi — CHANGES_REQUESTED** [review]
Thanks, looks good except for the typo. can you fix that?

**Comment by klausi** [comment]
PHPstan tests are failing here, can you merge in the 8.x-3.x branch?

File: coder_sniffer/Drupal/Sniffs/Functions/DiscouragedFunctionsSniff.php (hunk 1 above)
Comment (Nicasso) [reply coder_sniffer/Drupal/Sniffs/Functions/DiscouragedFunctionsSniff.php]: Ah, my bad. Hold on!

Comment (Nicasso) [reply coder_sniffer/Drupal/Sniffs/Functions/DiscouragedFunctionsSniff.php]: Fixed

**Comment by Nicasso** [comment]
Hi Klausi,

As far as I can see I'm up-to-date with the 8.x-3.x branch.

**Comment by klausi** [comment]
phpstan tests fails are on the 8.x-3.x branch, fixed them there.

**Pull request merged**